components/storage/blockdevice/COMPONENT_SD/SDBlockDevice.h

```diff
@@ -192,6 +192,15 @@ class SDBlockDevice : public mbed::BlockDevice {
      */
     virtual const char *get_type() const;
 
+    /** Get the BlockDevice initialization status.
+     *
+     * @return the private variable _init_ref_count if it was bigger than 0
+     */
+    virtual bool IsInitialized()
```

Comment:
This is an attempt to fix a bug by extending the public API.

When the application is written correctly, init and de-init calls match, and the driver gets de-initialized correctly.

If you don't know, whether there are other users of this driver, meaning `_init_ref_count` does not reach zero, you also should not force the deinitialization of the driver, breaking the other user.

So, I would say that we do not merge this change in.

Reply:
Please refer to this case 
https://forums.mbed.com/t/detrmining-mounting-status-of-sdblockdevice/7575

Reply:
When `SDBlockDevice::init()` fails, it returns an error code.

If you want to try retrying, you need to call `SDBlockDevice::deinit()`, before another `SDBlockDevice::init()`.

API works as expected. `init()` and `deinit()` calls much match. Otherwise, you cannot chain and combine blockdevices properly.

Reply:
I don't think so as I can capture that the variable is incremented without I do that. 
Even if there is not any problem, you need to know the status of the initialization ... and there is no interface/API/implementation for that